flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesSharedInformer.java

```diff
@@ -102,18 +91,31 @@ public Watch watch(
 
     @Override
     public void run() {
-        sharedInformerFactory.startAllRegisteredInformers();
+        startInformerAsync(sharedIndexInformer, aggregatedEventHandler);
     }
 
     @Override
     public void close() {
-        sharedInformerFactory.stopAllRegisteredInformers();
+        this.sharedIndexInformer.stop();
+        ExecutorUtils.gracefulShutdown(5, TimeUnit.SECONDS, this.informerExecutor);
     }
 
     private String getResourceKey(String name) {
         return client.getNamespace() + "/" + name;
     }
 
+    private void startInformerAsync(
+            SharedIndexInformer<T> informer, SharedInformerEventListener eventListener) {
+        informerExecutor.execute(
```

Comment:
Why do need the executor for this? Is `run` such a heavy operation?

Reply:
Looks like we don't need run informer here, it has been run while invoking `Informable#inform`

Reply:
@yittg Thanks for your hint. I think you are right. We do not need to start the informer anymore.

Since `Informable#inform`, including run the informer internally, is not very heavy, I will keep it in the main thread. After then, the exception will also forward to the caller thread. We do not need to handle the exception in the `AggregatedEventHandler`.